dynamiqs/mesolve/rouchon.py

```diff
@@ -115,51 +99,44 @@ def backward_augmented(
         # rho: (b_H, b_rho, n, n) -> (b_H, b_rho, n, n)
         # phi: (b_H, b_rho, n, n) -> (b_H, b_rho, n, n)
 
-        # compute cached operators
-        # H, Hnh, M0, M0rev: (b_H, 1, n, n)
         H = self.H(t)
         Hnh = self.Hnh(H)
-        M0 = self.M0(Hnh)
-        M0rev = self.M0rev(Hnh)
+
+        # === reverse time
+        M0rev, M1srev = self.Ms(Hnh, -self.dt)
 
         # normalize the Kraus Map
-        if self.cholesky_normalization:
-            # Rrev, Trev: (b_H, 1, n, n)
-            Rrev = self.Rrev(M0rev)
-            Trev = self.Trev(Rrev)
+        if self.options.normalize == 'cholesky':
+            Rrev = self.R(M0rev, -self.dt)
+            Trev = self.T(Rrev)
             rho = inv_kraus_matmul(Trev.mH, rho, upper=True)  # Tr.mH^-1 @ rho @ Tr^-1
 
         # compute rho(t-dt)
-        rho = kraus_map(rho, M0rev) - kraus_map(rho, self.M1srev)
+        rho = kraus_map(rho, M0rev) - kraus_map(rho, M1srev)
```

Comment:
If you use a `-dt` in the Kraus operator, then this should be `rho = kraus_map(rho, M0rev) + kraus_map(rho, M1srev)`